sensors/sqs_sensor.py

```diff
@@ -98,50 +111,117 @@ def _may_setup_sqs(self):
         else:
             self.input_queues = []
 
-        self.aws_access_key = self._get_config_entry('aws_access_key_id')
-        self.aws_secret_key = self._get_config_entry('aws_secret_access_key')
-        self.aws_region = self._get_config_entry('region')
+        # checker configuration is update, or not
+        def _is_same_credentials(session, account_id):
+            c = session.get_credentials()
+            same_credentials = c is not None and \
+                c.access_key == self.credentials[account_id][0] and \
+                c.secret_key == self.credentials[account_id][1]
+            if account_id != self.account_id:
+                return same_credentials and c.token == self.credentials[account_id][2]
+            else:
+                return same_credentials
 
-        self.max_number_of_messages = self._get_config_entry('max_number_of_messages',
-                                                             prefix='sqs_other')
+        cross_roles_arns = {
+            arn.split(':')[4]: arn for arn in self._get_config_entry('roles_arns', 'sqs_sensor') or []
+        }
+        required_accounts = {self._get_info(queue)[0] for queue in self.input_queues}
 
-        # checker configuration is update, or not
-        def _is_same_credentials():
-            c = self.session.get_credentials()
-            return c is not None and \
-                c.access_key == self.aws_access_key and \
-                c.secret_key == self.aws_secret_key and \
-                self.session.region_name == self.aws_region
-
-        if self.session is None or not _is_same_credentials():
-            self._setup_sqs()
-
-    def _setup_sqs(self):
-        ''' Setup Boto3 structures '''
-        self._logger.debug('Setting up SQS resources')
-        self.session = Session(aws_access_key_id=self.aws_access_key,
-                               aws_secret_access_key=self.aws_secret_key,
-                               region_name=self.aws_region)
+        for account_id in required_accounts:
+            if account_id != self.account_id and account_id not in cross_roles_arns:
+                continue
+
+            session = self.sessions.get(account_id, None)
```

Comment:
I think `None` is default, so you can have `session = self.sessions.get(account_id)`